pkg/buildchange/change_processor.go

```diff
@@ -0,0 +1,111 @@
+package buildchange
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+)
+
+const (
+	reasonsSeparator = ","
+	errorSeparator   = "\n"
+)
+
+func NewChangeProcessor() *ChangeProcessor {
+	return &ChangeProcessor{
+		changes: []GenericChange{},
+		errStrs: []string{},
+	}
+}
+
+type ChangeProcessor struct {
+	changes []GenericChange
+	errStrs []string
+}
+
+func (c *ChangeProcessor) Process(change Change) *ChangeProcessor {
+	if change == nil {
+		return c
+	}
+
+	if !change.Reason().IsValid() {
+		errStr := fmt.Sprintf("unsupported change reason '%s'", change.Reason())
+		c.errStrs = append(c.errStrs, errStr)
+		return c
+	}
+
+	valid, err := change.IsValid()
```

Comment:
While reading this PR, I kept getting a bit confused by the term `isValid`. It seems that `IsValid` is essentially asking if the change is necessary for a new build. Could we do something like `RequiresRebuild` or something?

Reply:
resolved by this commit - https://github.com/pivotal/kpack/pull/556/commits/3a47900f134f03876914ae0a94788698dd7dca45